File: software/zyarm_sdk/python/examples/read_state.py
from zyarm_sdk import ZyArm, ZyArmConfig


def main() -> None:
    with ZyArm(ZyArmConfig(port="/dev/ttyUSB0")) as arm:
        state = arm.query_state(timeout_ms=1000)
        if state is None:
            print("No fresh state received")
            return
        print(state.positions)


if __name__ == "__main__":
    main()

File: software/zyarm_sdk/cpp/examples/read_state.cpp
#include <iostream>

#include "zyarm_sdk/arm.hpp"

int main()
{
  zyarm_sdk::ZyArmConfig config;
  config.port = "/dev/ttyUSB0";
  zyarm_sdk::ZyArm arm(config);
  arm.connect();
  auto state = arm.query_state(std::chrono::milliseconds(1000));
  if (!state.has_value()) {
    std::cout << "No fresh state received\n";
    return 1;
  }
  for (double value : state->positions) {
    std::cout << value << " ";
  }
  std::cout << "\n";
  arm.close();
  return 0;
}
